src/dos/drive_local.cpp

```diff
@@ -443,6 +480,18 @@ bool localDrive::GetFileAttr(char *name, uint16_t *attr)
 			*attr |= DOS_ATTR_DIRECTORY;
 		if (!(status.st_mode & S_IWUSR))
 			*attr |= DOS_ATTR_READ_ONLY;
+
+		char xattr[XATTR_FATTR_LEN] = "";
+#ifdef MACOSX
+		int read_len = getxattr(newname, XATTR_FATTR_NAME, xattr, XATTR_FATTR_LEN, 0, 0);
+#else
+		int read_len = getxattr(newname, XATTR_FATTR_NAME, xattr, XATTR_FATTR_LEN);
```

Comment:
This should be `XATTR_FATTR_LEN - 1` so you leave room to add a null terminator.